osquery/tables/networking/posix/lldp_neighbors.cpp

```diff
@@ -0,0 +1,417 @@
+                  LLDP_DOT3_LINK_AUTONEG_1000BASE_T,
+                  LLDP_DOT3_LINK_AUTONEG_1000BASE_TFD,
+                  "port_autoneg_1000baset_hd_enabled",
+                  "port_autoneg_1000baset_fd_enabled");
+  }
+}
+
+void LLDPNeighbor::getChassis() {
+  row_["rid"] = getAtomStr(chassis_, lldpctl_k_chassis_index);
+  row_["chassis_id"] = getAtomStr(chassis_, lldpctl_k_chassis_id);
+  row_["chassis_id_type"] = getAtomStr(chassis_, lldpctl_k_chassis_id_subtype);
+  row_["chassis_sysname"] = getAtomStr(chassis_, lldpctl_k_chassis_name);
+  row_["chassis_sys_description"] =
+      getAtomStr(chassis_, lldpctl_k_chassis_descr);
+
+  getChasisCapability(LLDP_CAP_BRIDGE,
+                      "chassis_bridge_capability_available",
+                      "chassis_bridge_capability_enabled");
+  getChasisCapability(LLDP_CAP_ROUTER,
+                      "chassis_router_capability_available",
+                      "chassis_router_capability_enabled");
+  getChasisCapability(LLDP_CAP_WLAN,
+                      "chassis_wlan_capability_available",
+                      "chassis_wlan_capability_enabled");
+  getChasisCapability(LLDP_CAP_REPEATER,
+                      "chassis_repeater_capability_available",
+                      "chassis_repeater_capability_enabled");
+  getChasisCapability(LLDP_CAP_TELEPHONE,
+                      "chassis_tel_capability_available",
+                      "chassis_tel_capability_enabled");
+  getChasisCapability(LLDP_CAP_DOCSIS,
+                      "chassis_docsis_capability_available",
+                      "chassis_docsis_capability_enabled");
+  getChasisCapability(LLDP_CAP_STATION,
+                      "chassis_station_capability_available",
+                      "chassis_station_capability_enabled");
+  getChasisCapability(LLDP_CAP_OTHER,
+                      "chassis_other_capability_available",
+                      "chassis_other_capability_enabled");
+
+  std::unique_ptr<lldpctl_atom_t, decltype(delLLDPAtom)> mgmts(
+      lldpctl_atom_get(chassis_, lldpctl_k_chassis_mgmt), delLLDPAtom);
+  row_["chassis_mgmt_ips"] = commaDelimitedStr(mgmts.get(), lldpctl_k_mgmt_ip);
+}
+
+/**
+ * @brief getNeighbor retreives all LLDP information of the given neighbor port
+ * and chassis.
+ *
+ * @return filled osquery::Row of lldp information for a given interface
+ *
+ */
+Row& LLDPNeighbor::getNeighbor() {
+  getChassis();
+  getPort();
+  getVLAN();
+  getMED();
+  getMEDCap();
+  getPIDs();
+  getPPVIDs();
+
+  return row_;
+}
+
+QueryData genLLDPNeighbors(QueryContext& context) {
+  QueryData rows;
+
+  auto delConn = [](lldpctl_conn_t* c) { lldpctl_release(c); };
+  std::unique_ptr<lldpctl_conn_t, decltype(delConn)> conn(
+      lldpctl_new(nullptr, nullptr, nullptr), delConn);
+
+  lldpctl_error_t err = lldpctl_last_error(conn.get());
+  if (err != LLDPCTL_NO_ERROR) {
+    LOG(ERROR) << "could not initiate new lldpd connection: "
+               << lldpctl_strerror(err);
+    return rows;
```

Comment:
I'm not sure what you mean...the lldpctl_conn_t pointer is now a unique_ptr and should be de-reffed when the pointer goes out of scope..

If it's regards to how the lldpctl_error_t returned by lldpctl_last_error, there's no mention of it it the docs of it being referenced counted.  Also I did a leak test with valgrind whilst hitting the code path [here](https://github.com/digitalocean/osquery/blob/ec6b9cfeca7eaa77be11667e6c8648a18c598ce2/osquery/tables/networking/posix/lldp_neighbors.cpp#L386) and didn't see any lost blocks being reported.